src/subsystems/nodejs/builders/strict-builder/nodejs_builder/nodejs_builder/lib/dependencies.py

```diff
@@ -0,0 +1,135 @@
+from dataclasses import dataclass
+from typing import Any, Callable, Literal, Optional, TypedDict, Union
+
+from .logger import logger
+
+
+@dataclass
+class Dependency:
+    name: str
+    version: str
+    derivation: str
+    parent: Union["Dependency", None] = None
+    dependencies: Union[dict[str, Any], None] = None
+
+    def repr(self: "Dependency") -> str:
+        return f"{self.name}@{self.version}"
```

Comment:
```suggestion
    def __str__(self: "Dependency") -> str:
        return f"{self.name}@{self.version}"
```

see https://docs.python.org/3/reference/datamodel.html?highlight=__str__#object.__str__